### PR TITLE
Newest Changes

### PR DESCRIPTION
**Pull request by juanmc2005**
## Biggest Changes
- Added STS Evaluation
- Refactored Evaluators and Datasets
- Speaker Verification Bugfixes
- Contrastive loss for STS
- Softmax loss for STS

### CODE REVIEW

File: losses/contrastive.py [hunk 1]
```diff
@@ -29,16 +33,26 @@ def forward(self, feat, logits, y):
...
+            # Calculate the ground truth Y corresponding to the pairs
+            gt = []
+            for i in range(nbatch-1):
+                for j in range(i+1, nbatch):
+                    gt.append(int(y[i] != y[j]))
+            gt = torch.Tensor(gt).float().to(self.device)
+            # Calculate the losses as described in the paper
+            loss = (1-gt) * torch.pow(dist, 2) + gt * torch.pow(torch.clamp(self.margin - dist, min=1e-8), 2)
+            # Normalize by batch size
+            return torch.sum(loss) / 2 / dist.size(0)
```
Comment (juanmc2005) [inline losses/contrastive.py]: `size_average` should be used here too
